buildSrc/src/main/java/quilt/internal/tasks/EnigmaProfileConsumingTask.java

```diff
@@ -0,0 +1,49 @@
+package quilt.internal.tasks;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Internal;
+import org.quiltmc.enigma.api.EnigmaProfile;
+import org.quiltmc.enigma.api.service.JarIndexerService;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import static org.quiltmc.enigma_plugin.Arguments.SIMPLE_TYPE_FIELD_NAMES_PATH;
+
+public abstract class EnigmaProfileConsumingTask extends DefaultMappingsTask {
+    @Internal(
+        "An EnigmaProfile cannot be fingerprinted. " +
+            "Up-to-date-ness is ensured by getSimpleTypeFieldNamesFiles and its source, " +
+            "MappingsExtension::getEnigmaProfileFile."
```

Comment:
Is `MappingsExtension::getEnigmaProfileFile` ever actually marked/added as a task input? Or is the only stuff from it that can affect stuff the simpleTypeFieldNamesFiles?